dialer.go

```diff
@@ -267,8 +267,14 @@ func (d *customDialer) getNetworkType(network string) string {
 func (d *customDialer) writeWARCFromConnection(ctx context.Context, reqPipe, respPipe *io.PipeReader, scheme string, conn net.Conn) {
 	defer d.client.WaitGroup.Done()
 
+	// Check if a feedback channel has been provided in the context
+	var feedbackChan chan struct{}
```

Comment:
Any reason for the chan not to be buffered of 1 so that the send doesn't block if not read instantly?